lib/board.rb

```diff
@@ -2,32 +2,58 @@ class Board
   attr_reader :grid
 
   def initialize
-    @grid = top_row
+    @grid = numbered_row
     build_grid
-    add_numbers
   end
 
   def to_s
-    grid.to_s
+    grid
   end
 
   private
 
-  def row
-    "|  |  |  |\n|__|__|__|\n".freeze
+  def numbered_row
+    "   1  2  3\n".freeze
+  end
+
+  def build_grid
+    add_top_row
+    add_rows
+  end
+
+  def add_top_row
+    @grid += top_row
   end
 
   def top_row
-    " __ __ __\n".freeze
+    "   __ __ __\n".freeze
   end
 
-  def build_grid
-    3.times do
-      @grid += row
-    end
+  def add_rows
+    @grid += rows
   end
 
-  def add_numbers
+  def rows
+    first_labeled_row + second_labeled_row + third_labeled_row
+  end
 
+  def first_labeled_row
+    column_labels[0] + row
```

Comment:
There's still a bit of a focus on the array here, in that three different methods know about it and how to access data within it. What could things look like if I asked you to not use an array (or an underlying data structure) at all?

Reply:
haha, hmm maybe I'm resisting something like 
```def first_labeled_row
   "A" + row
end
```
because it feels "hardcoded" and any change would have to update all three methods instead of the array. On the other hand, it is easier to read and is what we're building here.

Reply:
Tried it out here 842a1f916ab0471db4cbee4d94a18baf94b19751

Reply:
I'd argue that the array literal is also hardcoded and would take as much effort to change as the example above.

Another thing to consider is that you can see the "backlog" and that the labels aren't intended to change. How much do you want a hypothetical feature change to affect your design?

What would you if your example looked like `row("A")`?

Reply:
What do you mean about backlog?

> What would you think if your example looked like row("A")

 I like that design a lot, once I committed I noticed that it's pretty redundant too.

Reply:
I was cheekily referring to the numbered list in the readme.

Reply:
heh got it!